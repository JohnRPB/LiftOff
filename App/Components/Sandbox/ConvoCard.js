import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Image, TouchableOpacity, Text, View } from 'react-native'
import ExamplesRegistry from '../../Services/ExamplesRegistry'

// Import all the pieces
// import ConvoBanner from './ConvoBanner';
import FadedSectionTitle from './FadedSectionTitle';
import RoundedButton from '../RoundedButton';

export default class ConvoCard extends Component {
  static propTypes = {
    anchorImage: PropTypes.object,
    anchorName: PropTypes.string,
    headShot: PropTypes.number.isRequired,
    bannerImage: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    knowledgeDomains: PropTypes.array.isRequired,
  }

  render() {
    console.log("------------------- START this.props.bannerImage -------------------");
    console.log(this.props.bannerImage);
    console.log("-------------------- END this.props.bannerImage --------------------");
    return (
      <View style={style.overall}>
        <Image source={this.props.bannerImage} style={style.banner} />
        <View style={style.content}>
          <Text style={style.title}> 
            {this.props.title} 
          </Text>
          <View style={{
            flexDirection:'row',
            justifyContent:'flex-start',
            }} >
            <Image source={this.props.headShot} style={style.headShot} />
            <Text> with Sam Harris </Text>
          </View>
          <Text style={style.convoSummary}>{this.props.summary}</Text> 
          <View style={{
            height: 120,
            flexDirection:'row', 
            flexWrap:'wrap',
            alignContent: 'stretch',
            paddingTop: 10,
            paddingBottom: 10,
          }}>
            <View style={style.smallSection}>
              <FadedSectionTitle>WHEN</FadedSectionTitle>
              <Text>{this.props.date}</Text>
            </View>
            <View style={style.smallSection}>
              <FadedSectionTitle>WHERE</FadedSectionTitle>
              <Text>{this.props.location}</Text>
              </View>
            <View style={style.smallSection}>
              <FadedSectionTitle>KNOWLEDGE DOMAINS</FadedSectionTitle>
              <Text>{this.props.knowledgeDomains.join(', ')}</Text>
            </View>
          </View>
          <RoundedButton buttonStyle={style.roundedButton}> Request to join </RoundedButton>
        </View>
      </View>
    )
  }
}

let style = {
  overall: {
    height: 541,
    width: 334,
    flexDirection: 'column',
    alignItems: 'center',
    borderRadius: 16.5,
    backgroundColor: "#ffffff",
    shadowColor: "rgba(0, 0, 0, 0.25)",
    shadowOffset: {
      width: 0,
      height: 0.5
    },
    shadowRadius: 2,
    shadowOpacity: 1,
    overflow:'hidden'
  },
  content: {
    height: 300,
    width: 280,
    margin: 32,
    flexDirection: 'column', 
    justifyContent: 'space-around',
  },
  banner: {
    borderColor:'black',
    justifyContent: 'center',
    alignSelf:'flex-start',
    width: 334,
    height: 161
  },
  headShot: {
    borderRadius: 12.5,
    marginRight: 5,
    width: 25,
    height: 25,
  },
  title: {
    opacity: 0.75,
    fontFamily: "SourceSansPro",
    fontSize: 20,
    fontWeight: "bold",
    fontStyle: "normal",
    letterSpacing: -0.1,
    color: "#000000"
  },
  convoSummary: {
    fontFamily: "SourceSansPro",
    fontSize: 16,
    fontWeight: "normal",
    fontStyle: "normal",
    letterSpacing: -0.23,
    color: "rgba(0, 0, 0, 0.5)"
  },
  smallSection: {
    paddingRight: 20
  },
  roundedButton: {
    borderRadius: 8.8,
    width:270,
    height:40,
    backgroundColor: "#373737",
    shadowColor: "rgba(0, 0, 0, 0.05)",
    shadowOffset: {
      width: 0,
      height: 0
    },
    shadowRadius: 2,
    shadowOpacity: 1
  }
}

// Ignore in coverage report
/* istanbul ignore next */
//ExamplesRegistry.addComponentExample('Conversation Card', () =>
  //<ConvoCard
    //bannerImage={Images.civilityLogo}
    //title='The Origins of Consciousness'
    //summary='Lets talk about the scientic study of consciousness, where consciousness emerges in nature, and more.'
    //date='Wed, 1/25,4-7pm'
    //location='San Francisco'
    //knowledgeDomains={['consciousness','psychology','psychedelics']}
  ///>
//)
