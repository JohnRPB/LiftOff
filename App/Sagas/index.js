import { takeLatest, all } from 'redux-saga/effects'
import githubAPI from '../Services/Api'
import FixtureAPI from '../Services/FixtureApi'
import FixtureLoginAPI from '../Services/FixtureLoginApi'
import DebugConfig from '../Config/DebugConfig'
import FixtureLoginApi from '../Services/FixtureLoginApi';

/* ------------- Types ------------- */

import { StartupTypes } from '../Redux/StartupRedux'
import { GithubTypes } from '../Redux/GithubRedux'
import { LoginTypes } from '../Redux/LoginRedux'

/* ------------- Sagas ------------- */

import { startup } from './StartupSagas'
import { getUserAvatar } from './GithubSagas'
import { loginUser } from './LoginSagas';

/* ------------- API ------------- */

// The API we use is only used from Sagas, so we create it here and pass along
// to the sagas which need it.
const githubApi = DebugConfig.useFixtures ? FixtureAPI : githubAPI.create()
const loginApi = FixtureLoginAPI

/* ------------- Connect Types To Sagas ------------- */

export default function * root () {
  yield all([
    // some sagas only receive an action
    takeLatest(StartupTypes.STARTUP, startup),

    // some sagas receive extra parameters in addition to an action
    takeLatest(GithubTypes.USER_REQUEST, getUserAvatar, githubApi),
    takeLatest(LoginTypes.LOGIN_REQUEST, loginUser, loginApi)
  ])
}
