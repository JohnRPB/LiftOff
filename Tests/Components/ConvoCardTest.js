import 'react-native'
import React from 'react'
import ConvoCard from '../../App/Components/Sandbox/ConvoCard'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import RoundedButton from '../../App/Components/RoundedButton';
import { Images } from '../../App/Themes';
import { View } from 'react-native'



test('ConvoCard component renders correctly', () => {
  const tree = renderer.create(
    <View style={{backgroundColor: "#f2f2f2", padding:10}}>
      <ConvoCard
        bannerImage={Images.samHarrisConvo}
        headShot={Images.samHarris}
        title='The Origins of Consciousness'
        summary='Lets talk about the scientic study of consciousness, where consciousness emerges in nature, and more...'
        date='Wed, 1/25, 4-7pm'
        location='San Francisco'
        knowledgeDomains={['consciousness','psychology','psychedelics']}
      />
    </View>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
