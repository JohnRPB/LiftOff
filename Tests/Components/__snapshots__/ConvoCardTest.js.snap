// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConvoCard component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 16.5,
        "flexDirection": "column",
        "height": 541,
        "overflow": "hidden",
        "shadowColor": "rgba(0, 0, 0, 0.25)",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 334,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "alignSelf": "flex-start",
          "borderColor": "black",
          "height": 161,
          "justifyContent": "center",
          "width": 334,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "column",
          "height": 300,
          "justifyContent": "space-around",
          "margin": 32,
          "width": 280,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "SourceSansPro",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "letterSpacing": -0.1,
            "opacity": 0.75,
          }
        }
      >
        The Origins of Consciousness
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "borderRadius": 12.5,
              "height": 25,
              "marginRight": 5,
              "width": 25,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
           with Sam Harris 
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontFamily": "SourceSansPro",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "letterSpacing": -0.23,
          }
        }
      >
        Lets talk about the scientic study of consciousness, where consciousness emerges in nature, and more...
      </Text>
      <View
        style={
          Object {
            "alignContent": "stretch",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 120,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "paddingRight": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontFamily": "SourceSansPro",
                "fontSize": 12.5,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0.34,
                "opacity": 0.15,
              }
            }
          >
            WHEN
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Wed, 1/25, 4-7pm
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingRight": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontFamily": "SourceSansPro",
                "fontSize": 12.5,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0.34,
                "opacity": 0.15,
              }
            }
          >
            WHERE
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            San Francisco
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingRight": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontFamily": "SourceSansPro",
                "fontSize": 12.5,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0.34,
                "opacity": 0.15,
              }
            }
          >
            KNOWLEDGE DOMAINS
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            consciousness, psychology, psychedelics
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#373737",
            "borderRadius": 8.8,
            "height": 40,
            "opacity": 1,
            "shadowColor": "rgba(0, 0, 0, 0.05)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 270,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginVertical": 10,
              "textAlign": "center",
            }
          }
        >
           Request to join 
        </Text>
      </View>
    </View>
  </View>
</View>
`;
